addon/validators/confirmation.js

```diff
@@ -1,22 +1,9 @@
-import Ember from 'ember';
 import buildMessage from 'ember-changeset-validations/utils/validation-errors';
-
-const {
-  get,
-  isPresent,
-  isEqual,
-  isEmpty,
-} = Ember;
+import { validate } from 'ember-validators';
 
 export default function validateConfirmation(options = {}) {
-  let { on, allowBlank } = options;
-
   return (key, newValue, _oldValue, changes/*, _content*/) => {
-    if (allowBlank && isEmpty(newValue)) {
-      return true;
-    }
-
-    return isPresent(newValue) && isEqual(get(changes, on), newValue) ||
-      buildMessage(key, 'confirmation', newValue, options);
+    let result = validate('confirmation', newValue, options, changes, key);
+    return (result === true) ? true : buildMessage(key, result);
```

Comment:
@poteto that wouldnt work since `validate` will always return either `true` or a `ValidationError` object which will always be a truthy value.